src/KM_FormMain.pas

```diff
@@ -436,12 +436,21 @@ procedure TFormMain.RGPlayerClick(Sender: TObject);
 
 
 procedure TFormMain.chkSuperSpeedClick(Sender: TObject);
+var Speed: Single;
 begin
   if (gGameApp.Game = nil)
-  or (gGameApp.Game.IsMultiplayer and not MULTIPLAYER_SPEEDUP and not gGameApp.Game.IsReplay) then
+  or (gGameApp.Game.IsMultiplayer
+    and not gGameApp.Game.IsMPGameSpeedUpAllowed
+    and not MULTIPLAYER_SPEEDUP
+    and not gGameApp.Game.IsReplay) then
     Exit;
 
-  gGameApp.Game.SetGameSpeed(IfThen(chkSuperSpeed.Checked, 300, 1), False);
+  if gGameApp.Game.IsMPGameSpeedUpAllowed then
```

Comment:
Why is this check needed? We should have already exited if MP speed up is not allowed. And even if it isn't, DefaultMPGameSpeed is what we should set the default speed to anyway right?